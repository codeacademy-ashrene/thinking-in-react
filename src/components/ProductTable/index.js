import React, { Component } from 'react';
import ProductCategoryRow from '../ProductCategoryRow';
import ProductRow from '../ProductRow';
import PropTypes from 'prop-types';
const groupByCategory = require( '../../utils/groupByCategory');


class ProductTable extends Component {
  render() {
    const rows = [];
    const groupedByCategory = groupByCategory(this.props.products);
    Object.keys(groupedByCategory).forEach((keyCategory) => {
      rows.push(<ProductCategoryRow category={keyCategory}  key={keyCategory}/>);
      groupedByCategory[keyCategory].forEach((productElement) => {
        if(productElement[0].indexOf(this.props.filterText) === -1 ){
          return;
        }
        if(this.props.inStockOnly && !productElement[2]){
          return;
        }
        rows.push(<ProductRow products={
          {name: productElement[0], 
          stocked: productElement[2], 
          price: productElement[1]}
        } key = {productElement[0]}/>);
      });
    });
    return (
      <table>
        <thead>
          <tr>
            <th>Name</th>
            <th>Price</th>
          </tr>
        </thead>
        <tbody>
          {rows}
        </tbody>
      </table>
    );
}
}

ProductTable.propTypes = {
  products: PropTypes.array.isRequired,
  filterText: PropTypes.string,
  inStockOnly: PropTypes.bool,
};

ProductTable.defaultProps = {
  filterText: '',
  inStockOnly: false,
}

export default ProductTable;