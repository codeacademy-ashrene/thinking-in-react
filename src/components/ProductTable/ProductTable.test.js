import React from 'react';
import ProductTable from '.';
import renderer from 'react-test-renderer';

describe('ProductTable', () => {
  const stockedProduct=[{
    category: 'Sporting Goods',
    name: 'Football',
    stocked: false,
    price: 3.99
  }];
  it('should render product table', () => {
    const tree = renderer.create(<ProductTable products={stockedProduct}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
})