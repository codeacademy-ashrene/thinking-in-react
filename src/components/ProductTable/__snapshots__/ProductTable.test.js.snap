// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTable should render product table 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Price
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>
        Sporting Goods
      </th>
    </tr>
    <tr>
      <td>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
          Football
        </span>
      </td>
      <td>
        3.99
      </td>
    </tr>
  </tbody>
</table>
`;
